### ISSUE TITLE
Bug report, the missing logits in save_gradient function?

### ISSUE THREAD

**Issue by machanic** [dydjw9/MetaAttack_ICLR2020#3]
In `save_gradient` function
https://github.com/dydjw9/MetaAttack_ICLR2020/blob/master/grad_gen/cifar_gen_grad_for_meta/utils.py#L169
in this line ,you just save `process_data[str(batch_idx)] = [data, grad, target]`.
But in data reader python,  for example, in this line:
https://github.com/dydjw9/MetaAttack_ICLR2020/blob/master/meta_training/imagenet_meta_training/imagenet.py#L60
How should you read `logits` that have been **never** stored to `npy` file before?

**Comment by machanic** [comment]
@xiaofanustc Can you help me for this question? thank you

**Comment by xiaofanustc** [comment]
sorry for the trouble, i just test the code on MNIST before uploading. Logits are not used finally and are removed now, the code is updated.

**Comment by machanic** [comment]
@xiaofanustc please check this code carefully again, because:
1. https://github.com/dydjw9/MetaAttack_ICLR2020/blob/master/meta_training/cifar_meta_training/meta.py#L35
The logits is still used
2. https://github.com/dydjw9/MetaAttack_ICLR2020/blob/master/meta_training/imagenet_meta_training/imagenet.py#L59
The logits is still read

**Comment by xiaofanustc** [comment]
should be fine this time, otherwise i also need to generate the gradient since some data are not in the current computer. So sorry about the inconvenience.

**Comment by machanic** [comment]
@xiaofanustc 
https://github.com/dydjw9/MetaAttack_ICLR2020/blob/master/meta_training/cifar_meta_training/meta.py#L124 
This line is incorrect after your update, because `l_qry` does not exist.

**Comment by xiaofanustc** [comment]
if you don't mind, i will run again tonight and give the revised version tomorrow. Thank you.

**Comment by machanic** [comment]
@xiaofanustc Thank you , if you can run from generate gradient --> train meta model --> attack , that is all pipeline to verify your code is correct. That would be nice and the your code is also tested.
